query/query/query.py

```diff
@@ -277,6 +277,15 @@ async def on_shutdown(app):
     log.info("All tasks on shutdown have completed")
 
 
+async def on_shutdown(_):
+    # Filter the asyncio.current_task(), because if we await
+    # the current task we'll end up in a deadlock
+    remaining_tasks = [t for t in asyncio.all_tasks() if t is not asyncio.current_task()]
+    log.info(f"On shutdown request received, with {len(remaining_tasks)} remaining tasks")
+    await asyncio.wait(*remaining_tasks)
+    log.info("All tasks on shutdown have completed")
+
+
```

Comment:
This will conflict with our other on_shutdown definition: https://github.com/populationgenomics/hail/blob/d45746067d69b40df99ba91b64f5a81806bfbcca/query/query/query.py#L267-L286

Reply:
Fixed in: 3ad57ea

Reply:
Thanks! Just double-checking: is this now in sync with upstream, since your change got merged upstream as well?

Reply:
Yes, that's why there was a duplicate, because there was a modification to the original PR (about the unused variable). I'm surprised that change didn't get applied automatically, so I think this is more of a "merge conflict" that just didn't cause a conflict.

Reply:
Awesome, thank you!